src/server/string_family_test.cc

```diff
@@ -691,4 +691,33 @@ TEST_F(StringFamilyTest, SetWithGetParam) {
   EXPECT_EQ(Run({"get", "key2"}), "val3");
 }
 
+TEST_F(StringFamilyTest, SetWithHashtags) {
+  EXPECT_EQ(Run({"set", "{key}1", "val1"}), "OK");
+  if (KeyLockArgs::IsLockHashTagEnabled()) {
+    EXPECT_THAT(GetLastUsedKeys(), AllOf(Contains("key"), Not(Contains("{key}1"))));
+  } else {
+    EXPECT_THAT(GetLastUsedKeys(), AllOf(Contains("{key}1"), Not(Contains("key"))));
+  }
+
+  EXPECT_EQ(Run({"set", "{key}2", "val2"}), "OK");
+  if (KeyLockArgs::IsLockHashTagEnabled()) {
+    EXPECT_THAT(GetLastUsedKeys(), AllOf(Contains("key"), Not(Contains("{key}2"))));
+  } else {
+    EXPECT_THAT(GetLastUsedKeys(), AllOf(Contains("{key}2"), Not(Contains("key"))));
+  }
+
+  EXPECT_THAT(Run({"mget", "{key}1", "{key}2"}), RespArray(ElementsAre("val1", "val2")));
```

Comment:
how about adding a test case for multi transaction?

Reply:
Good idea, done!